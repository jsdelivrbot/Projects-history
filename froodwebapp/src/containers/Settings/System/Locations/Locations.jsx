/* eslint-disable class-methods-use-this, react/no-array-index-key */
import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { withRouter } from 'react-router-dom';
import { connect } from 'react-redux';
import { Row, Col } from 'react-flexbox-grid';
import {
  locationsGetRequest
} from 'redux-base/actions';
import { Tabs } from 'antd';
import { PageHeader } from 'components';
import {
  ConnectedLocationsTab,
  ConnectedZonesTab,
  ConnectedBinsTab
} from './Tabs';

const mapDispatchToProps = {
  locationsGetRequest
};

const { TabPane } = Tabs;

export class Locations extends Component {
  constructor(props) {
    super(props);
    this.state = {
      activeTab: 'locations'
    };

    if (this.props.location.state) {
      this.state = Object.assign({}, this.state, props.location.state);
    }
  }

  componentWillMount() {
    this.props.locationsGetRequest();
  }

  handleTabClick = (activeTab) => {
    this.setState({
      activeTab
    }, () => {
      switch (activeTab) {
        case 'locations':
          this.props.locationsGetRequest();
          break;
        default:
      }
    });
  }

  render() {
    const {
      activeTab,
      activeAddress
    } = this.state;

    return (
      <div>
        <PageHeader
          bigText="Company Locations"
          smallText="Manage company locations"
        />
        <Row>
          <Col lg>
            <Tabs
              activeKey={ activeTab }
              onTabClick={ this.handleTabClick }
              animated={ false }
            >
              <TabPane
                key="locations"
                tab="Locations"
              >
                <ConnectedLocationsTab activeAddress={ activeAddress } />
              </TabPane>
              <TabPane
                key="zones"
                tab="Zones"
              >
                <ConnectedZonesTab />
              </TabPane>
              <TabPane
                key="bins"
                tab="Bins"
              >
                <ConnectedBinsTab />
              </TabPane>
            </Tabs>
          </Col>
        </Row>
      </div>
    );
  }
}

Locations.propTypes = {
  // redux-base
  locationsGetRequest: PropTypes.func.isRequired,
  // router
  location: PropTypes.object.isRequired
};

export default withRouter(connect(null, mapDispatchToProps)(Locations));
