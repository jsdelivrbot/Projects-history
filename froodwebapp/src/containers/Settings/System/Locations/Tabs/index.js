export ConnectedBinsTab from './BinsTab/BinsTab';
export ConnectedLocationsTab from './LocationsTab/LocationsTab';
export ConnectedZonesTab from './ZonesTab/ZonesTab';
