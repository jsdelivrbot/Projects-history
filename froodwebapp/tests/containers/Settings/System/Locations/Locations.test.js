import { Locations } from 'containers/Settings/System/Locations/Locations';
import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

const { describe, expect, it } = global;

describe('Locations', () => {
  const props = {
    locationsGetRequest: jest.fn()
  };

  it('renders the Locations component', () => {
    const component = shallow(<Locations { ...props } />);

    expect(component.length).toEqual(1);
  });

  it('capturing Snapshot of Locations', () => {
    const component = shallow(<Locations { ...props } />);
    const view = toJson(component);
    const actual = expect(view);

    actual.toMatchSnapshot();
  });

  it('calls locationsGetRequest in handleTabClick when activeTab is locations', () => {
    const activeTab = 'locations';
    const component = shallow(<Locations { ...props } />);
    props.locationsGetRequest.mock.calls.length = 0;
    component.instance().handleTabClick(activeTab);

    expect(props.locationsGetRequest.mock.calls.length).toBe(1);
  });
});
