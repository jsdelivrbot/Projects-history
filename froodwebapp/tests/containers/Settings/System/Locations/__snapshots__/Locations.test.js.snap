// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Locations capturing Snapshot of Locations 1`] = `
<div>
  <PageHeader
    bigText="Company Locations"
    smallText="Manage company locations"
  />
  <Row>
    <Col
      lg={true}
    >
      <Tabs
        activeKey="locations"
        animated={false}
        hideAdd={false}
        onTabClick={[Function]}
        prefixCls="ant-tabs"
      >
        <TabPane
          key="locations"
          placeholder={null}
          tab="Locations"
        >
          <withRouter(Connect(LocationsTab)) />
        </TabPane>
        <TabPane
          key="zones"
          placeholder={null}
          tab="Zones"
        >
          <Connect(ZonesTab) />
        </TabPane>
        <TabPane
          key="bins"
          placeholder={null}
          tab="Bins"
        >
          <Connect(BinsTab) />
        </TabPane>
      </Tabs>
    </Col>
  </Row>
</div>
`;
